src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntryValidator.cs

```diff
@@ -134,6 +134,32 @@ private void ValidateCustomTagEntry(CustomTagEntry tagEntry)
             }
         }
 
+        private static void ValidatePrivateCreator(DicomTag tag, string privateCreator, string originalTagPath)
+        {
+            if (tag.IsPrivate)
```

Comment:
Are these checks we could have performed earlier in the class ctor?

Reply:
CustomTagEntry is used in this way:

HttpRequest Body => Deserialized by Asp.net Core to CustomTagEntry list =>Passed to AddCustomTags controller which does further validation.

So only the non-parameter constructor is used when deserialization.